src/Types/VersionedModuleSource.cs

```diff
@@ -92,6 +126,41 @@ versionedModuleSource.ModuleCase switch
             _ => throw new MissingEnumException<Grpc.V2.VersionedModuleSource.ModuleOneofCase>(versionedModuleSource
                 .ModuleCase)
         };
+
+    /// <summary>
+    /// Create a versioned module schema from a byte array.
+    /// </summary>
+    /// <param name="bytes">The serialized schema.</param>
+    /// <param name="output">Where to write the result of the operation.</param>
+    public static bool TryDeserial(ReadOnlySpan<byte> bytes, out (VersionedModuleSource? VersionedModuleSource, string? Error) output)
+    {
+        if (bytes.Length < VersionedModuleSource.MinSerializedLength)
+        {
+            output = (null, $"The given byte array in `VersionModuleSourceFactory.TryDeserial`, is too short. Must be longer than {VersionedModuleSource.MinSerializedLength}.");
+            return false;
+        }
+
+        var version = BinaryPrimitives.ReadUInt32BigEndian(bytes);
+        var length = BinaryPrimitives.ReadUInt32BigEndian(bytes[sizeof(int)..]);
```

Comment:
**Suggestion:** Again, we should have a comment here to why this is "safe" from exceptions